gazebo_ros/include/gazebo_ros/node.hpp

```diff
@@ -0,0 +1,153 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAZEBO_ROS_NODE_HPP
+#define GAZEBO_ROS_NODE_HPP
+
+#include <memory>
+#include <mutex>
+#include <atomic>
+
+#include <rclcpp/rclcpp.hpp>
+#include <gazebo/common/common.hh>
+
+#include <gazebo_ros/executor.hpp>
+
+namespace gazebo_ros
+{
+
+/// ROS Node for gazebo plugins
+/**
+ * \class Node node.hpp <gazebo_ros/node.hpp>
+ * Wrapper around an rclcpp::Node which ensures all instances share an exeuctor.
+ * \include gazebo_ros/node.hpp
+ */
+class Node : public rclcpp::Node
+{
+public:
+  /// Exception thrown when a #Create is called for a Node before #InitROS has ever been called
+  class NotInitializedException : public std::runtime_error
+  {
+    public:
+      NotInitializedException();
+  };
+
+  /// Shared pointer to a #gazebo_ros::Node
+  typedef std::shared_ptr<Node> SharedPtr;
+
+  /// Destructor
+  virtual ~Node();
+
+  /// Create a #gazebo_ros::Node and add it to the global #gazebo_ros::Executor.
+  /**
+   * \note Must NOT called until #InitROS has been called
+   * \param[in] node_name Name for the new node to create
+   * \return A shared pointer to a new #gazebo_ros::Node
+   */
+  static SharedPtr Create(const std::string& node_name);
```

Comment:
I *believe* the pointer style rule [here](https://github.com/ros2/ros2/wiki/Developer-Guide#pointer-syntax-alignment) also applies in this case, i.e. `std::string & node_name`. I also looked through existing ROS2 code and that seems to be the case.